modules/BrainCRISPR.py

```diff
@@ -0,0 +1,278 @@
+            re.sub(r"#+\s+", "", description_lines[0])
+            if "#" in description_lines[0]
+            else description_lines[0]
+        )
+
+        experiment = None
+        analysis = None
+
+        try:
+            for i in range(len(description_lines)):
+                if "## Experiment" == description_lines[i]:
+                    experiment = description_lines[i + 1]
+                if "## Analysis" == description_lines[i]:
+                    analysis = description_lines[i + 1]
+        except TypeError:
+            experiment = None
+            analysis = None
+
+        return {"title": title, "experiment": experiment, "analysis": analysis}
+
+    def __parse_gene_list(self: CRISPRBrain, study_id: str) -> DataFrame:
+        """
+        This function parses a gene list for a single screen and returns a dataframe with hit genes.
+
+        Args:
+          study_id (str): The study_id parameter is a string that represents the unique identifier for a
+        single screen.
+
+        Returns:
+          a DataFrame with hit genes from a single screen. The DataFrame contains columns
+            - targetFromSourceId,
+            - pValue,
+            - statisticalTestTail,
+            - resourceScore, and
+            - studyId.
+
+        The function filters out non-hit genes and renames some columns using the PySpark functions col and lit.
+        """
+        screen_url = f"https://storage.googleapis.com/crisprbrain.appspot.com/api-data/{quote(study_id)}.csv.gz"
+        self.spark.sparkContext.addFile(screen_url)
+
+        return (
+            self.spark.read.csv(
+                SparkFiles.get(f"{study_id}.csv.gz"), header=True, inferSchema=True
+            )
+            .filter(f.col("Hit Class") != "Non-Hit")
+            .select(
+                f.col("Gene").alias("targetFromSourceId"),
+                f.col("P Value").alias("resourceScore"),
+                f.when(f.col("Hit Class") == "Positive Hit", f.lit("upper tail"))
+                .when(f.col("Hit Class") == "Negative Hit", f.lit("lower tail"))
+                .alias("statisticalTestTail"),
+                f.col("Phenotype").alias("log2FoldChangeValue"),
+                f.lit(study_id).alias("studyId"),
+            )
+        )
+
+    def __get_all_hit_genes(self, study_identifiers: List[str]) -> DataFrame:
+        """Retrieving hit genes from all screens."""
+
+        # Parsing all gene lists for all screens and combine them into a single dataframe:
+        return reduce(
+            lambda df1, df2: df1.unionByName(df2),
+            map(self.__parse_gene_list, study_identifiers),
+        )
+
+    def create_crispr_brain_evidence(self: CRISPRBrain) -> None:
+        """"""
+        # Reading
+        crispr_studies = (
+            # Reading studies:
+            self.__get_study_table()
+            # Joining with resolved literature:
+            .join(self.__get_literature_mapping(), on="Reference Link", how="left")
```

Comment:
I see a potential problem here: if in the future they add more publications, the lut returned by `__get_literature_mapping()` will not be able to map them all, and will just _silently_ continue. I'm not sure what is the best solution here though. But perhaps displaying at least a warning would be a good idea?

Reply:
Any upcoming release will *likely* break the API:
- The [API key](https://github.com/opentargets/evidence_datasource_parsers/blob/bbc2ac9a486d70c0150603513205638e6d522c91/modules/BrainCRISPR.py#L24) will likely changed.
- Also the [API version](https://github.com/opentargets/evidence_datasource_parsers/blob/bbc2ac9a486d70c0150603513205638e6d522c91/modules/BrainCRISPR.py#L71) should be different as well. 

These checks are also implemented into the official wrapper.

Reply:
UPD: I now see below that the logging has already been implemented, now it's all good. Resolving this thread